### PR TITLE
implements APIM with internal vnet quickstart

### PR DESCRIPTION
**Pull request by kyleburnsdev**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 201-api-management-create-with-internal-vnet-application-gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Create API Management in Internal VNet with App Gateway
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/README.md:1]: move this sample to the quickstarts/Microsoft.ApiManagement folder and remove "201-" from the foldername

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+            "sku": {
+                "name": "[parameters('apim-sku')]",
+                "capacity": "[parameters('apim-capacity')]"
+            },
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "resources": [
+                {
+                    "type": "Microsoft.ApiManagement/service/gateways",
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:107]: since this is nested, just use "gateways" as the type

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+                    "apiVersion": "2020-12-01",
+                    "name": "[concat(variables('apim-name'), '/my-gateway')]",
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:109]: use "my-gateway" once the type is fixed

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+                            "response": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            }
+                        }
+                    }
+                },
+                {
+                    "type": "providers/diagnosticSettings",
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:180]: a few changes here:
1) move this to a top level resource
1) add the scope property
1) remove providers from the type

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+                        "metrics": [
+                            {
+                                "category": "AllMetrics",
+                                "enabled": true
+                            }
+                        ]
+                    }
+                }
+            ],
+            "dependsOn": [
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:204]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+            }
+        },
+        {
+            "type": "Microsoft.Network/applicationGateways",
+            "apiVersion": "2020-11-01",
+            "name": "[variables('app-gateway-name')]",
+            "location": "[parameters('location')]",
+            "resources": [
+                {
+                    "type": "providers/diagnosticSettings",
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:223]: apply the same changes here as in the above extension resource

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+                            "privateEndpointNetworkPolicies": "Enabled",
+                            "privateLinkServiceNetworkPolicies": "Enabled"
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Network/privateDnsZones",
+            "apiVersion": "2018-09-01", //was 2020-06-01
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:458]: remove comments

File: 201-api-management-create-with-internal-vnet-application-gateway/prereqs/.settings.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (bmoore-msft) [inline 201-api-management-create-with-internal-vnet-application-gateway/prereqs/.settings.json:1]: remove this - I don't see anything that requires this to be in the same RG

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 4 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:180]: I'm not sure that I follow on the syntax you're looking for here. I can move to top level resource, but the docs related to diagnosticSettings (https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/resource-manager-diagnostic-settings) shows needing a "type" following the pattern `resourceProvider/resourceType/providers/diagnosticSettings` and multi-segment name to appropriately associate the diagnostic setting with the correct resource. Is there an alternative syntax that is preferred here?

Comment (bmoore-msft) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json:180]: Here's an example:

https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.cache/redis-premium-cluster-diagnostics/azuredeploy.json#L95-L96

TLDR; https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/scope-extension-resources?tabs=azure-cli

Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Thanks for that, Brian. How to use the "scope" property was the key piece that I was missing and it makes sense now. I'll update to reflect this and suggest the doc that I linked be updated to reflect this practice as well

File: 201-api-management-create-with-internal-vnet-application-gateway/README.md (hunk 1 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/README.md]: Done

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 2 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Done - really helps readability

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 3 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Done

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 4 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Done

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 5 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Done - didn't catch "resources" as falling in the "everything not on this list goes before 'properties'"

File: 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json (hunk 7 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json]: Done

File: 201-api-management-create-with-internal-vnet-application-gateway/prereqs/.settings.json (hunk 8 above)
Comment (kyleburnsdev) [reply 201-api-management-create-with-internal-vnet-application-gateway/prereqs/.settings.json]: Done
